csv_detective/detection.py

```diff
@@ -122,42 +121,22 @@ def detect_separator(file, verbose: bool = False):
     return sep
 
 
-def detect_encoding(the_file, verbose: bool = False, unwanted=['ascii']):
+def detect_encoding(the_file, verbose: bool = False):
     """
-    Detects file encoding using charset_normalizer from lines
-    if feeding a bigger part of the file, detect most likely gets it wrong
-    (tried: detect(b"".join(lines[:min(len(lines), n)])) with n varying)
+    Detects file encoding using faust-cchardet (forked from the original cchardet)
     """
     if verbose:
         start = time()
         logging.info("Detecting encoding")
-    lines = the_file.readlines()
-    if len(lines) == 0:
-        raise ValueError("File is empty.")
-    elif len(lines) == 1:
-        encoding_this = detect(lines[0])['encoding']
-    # checking encoding for two-line batches until we get an agreement
-    else:
-        for k in range(1, len(lines)):
-            encoding_previous = detect(lines[k-1])['encoding']
-            encoding_this = detect(lines[k])['encoding']
-            if encoding_this not in unwanted and encoding_this == encoding_previous:
-                break
+    encoding_dict = detect(the_file.read())
```

Comment:
It's not a revert of the way we used to detect with cchardet before? https://github.com/etalab/csv-detective/pull/63/files#diff-69e4e16c9cb7c196a051417f601494e7963ce39ca9e1861f030d69f7d3823c0dL121

Reply:
Not exactly, [apparently](https://github.com/faust-streaming/cChardet#example) this is a correct way too

Reply:
I'm wondering if our previous implementation was not made to stop early? By feeding chunks by chunks.

Reply:
Seems like it, but I've tested both versions on a rather large file (the [first resource of [this dataset](https://www.data.gouv.fr/fr/datasets/bases-statistiques-communale-et-departementale-de-la-delinquance-enregistree-par-la-police-et-la-gendarmerie-nationales/) and the new method is substantially more efficient (4 times faster)